dotnet/src/Connectors/Connectors.Amazon/Bedrock/Extensions/BedrockKernelBuilderExtensions.cs

```diff
@@ -0,0 +1,575 @@
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (_, _)
+            => new BedrockChatCompletionService(modelId));
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using new AmazonBedrockRuntimeClient().
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <param name="region">The region to connect.</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        RegionEndpoint region,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (_, _)
+            => new BedrockChatCompletionService(modelId, region));
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using new AmazonBedrockRuntimeClient().
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <param name="clientConfig">The AmazonBedrockRuntimeClient Configuration Object</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        AmazonBedrockRuntimeConfig clientConfig,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (_, _)
+            => new BedrockChatCompletionService(modelId, clientConfig));
+
+        return builder;
+    }
+    /// <summary>
+    /// Add Amazon Bedrock Chat Completion service to the kernel builder using AWSCredentials object.
+    /// </summary>
+    /// <param name="builder">The kernel builder.</param>
+    /// <param name="modelId">The model for chat completion.</param>
+    /// <param name="awsCredentials">A credentials object for AWS services.</param>
+    /// <param name="serviceId">The optional service ID.</param>
+    /// <returns></returns>
+    public static IKernelBuilder AddBedrockChatCompletionService(
+        this IKernelBuilder builder,
+        string modelId,
+        AWSCredentials awsCredentials,
+        string? serviceId = null)
+    {
+        builder.Services.AddKeyedSingleton<IChatCompletionService>(serviceId, (_, _)
+            => new BedrockChatCompletionService(modelId, awsCredentials));
+
+        return builder;
+    }
```

Comment:
It might be better to keep only this extension method, and not provide other options.

Reply:
Received this update (from Norm): Have the methods take either the IAmazonBedrockRuntime or rely on the service client being configured via AWSSDK.Extensions.NETCore.Setup.

I'm keeping these two now:
```
/// <summary>
    /// Add Amazon Bedrock Chat Completion service to the kernel builder using IAmazonBedrockRuntime object.
    /// </summary>
    /// <param name="builder">The kernel builder.</param>
    /// <param name="modelId">The model for chat completion.</param>
    /// <param name="bedrockApi">The IAmazonBedrockRuntime to run inference using the respective model.</param>
    /// <param name="serviceId">The optional service ID.</param>
    /// <returns></returns>
    public static IKernelBuilder AddBedrockChatCompletionService(
        this IKernelBuilder builder,
        string modelId,
        IAmazonBedrockRuntime bedrockApi,
        string? serviceId = null)
    {
        builder.Services.AddSingleton<IChatCompletionService>(services =>
        {
            try
            {
                return new BedrockChatCompletionService(modelId, bedrockApi);
            }
            catch (Exception ex)
            {
                throw new KernelException($"An error occurred while initializing the BedrockChatCompletionService: {ex.Message}", ex);
            }
        });

        return builder;
    }

    /// <summary>
    /// Add Amazon Bedrock Chat Completion service to the kernel builder using new AmazonBedrockRuntimeClient().
    /// </summary>
    /// <param name="builder">The kernel builder.</param>
    /// <param name="modelId">The model for chat completion.</param>
    /// <param name="serviceId">The optional service ID.</param>
    /// <returns></returns>
    public static IKernelBuilder AddBedrockChatCompletionService(
        this IKernelBuilder builder,
        string modelId,
        string? serviceId = null)
    {
        // Add IAmazonBedrockRuntime service client to the DI container
        builder.Services.AddAWSService<IAmazonBedrockRuntime>();
        builder.Services.AddSingleton<IChatCompletionService>(services =>
        {
            try
            {
                var bedrockRuntime = services.GetRequiredService<IAmazonBedrockRuntime>();
                return new BedrockChatCompletionService(modelId, bedrockRuntime);
            }
            catch (Exception ex)
            {
                throw new KernelException($"An error occurred while initializing the BedrockChatCompletionService: {ex.Message}", ex);
            }
        });

        return builder;
    }
```

Would you recommend still keeping the method which takes AWSCredentials as well?